src/general.py

```diff
@@ -26,3 +34,132 @@ def get_resources() -> dict:
             unique_resources[resource["GLIDEIN_ResourceName"]] = 1
 
     return unique_resources
+
+
+def run_exerciser(tests_dir: Path, working_dir: Path, run=False):
+    """
+    Usage: calls necessary methods for setting up, running, and cleaning up exerciser
+    @param tests_dir: directory containing all exerciser tests
+    @param working_dir: directory for storing info on exerciser runs
+    @param run: whether or not to create working file system and run tests
+    """
+    if run:
+        execute_tests(tests_dir, working_dir)
+
+
+def execute_tests(tests_dir: Path, working_dir: Path):
+    """
+    Usage: builds working file system and submits tests
+    @param tests_dir: directory containing all exerciser tests
+    @param working_dir: directory for storing info on exerciser runs
+    """
+    # create top level working dir for exerciser run
+    curr_time = datetime.now().strftime("%Y-%m-%d_%H-%M")
+    timestamp_dir = os.path.join(working_dir, curr_time)
+    if not os.path.exists(timestamp_dir):
+        os.makedirs(timestamp_dir)
+
+        # create text file with list of currently available ResourceNames
+        with open(
+            os.path.join(timestamp_dir, "resource_list.txt"), "w"
+        ) as resource_list:
+            resources = get_resources()
+            for resource in resources.keys():
+                resource_list.write(f"{resource}\n")
+    else:
+        # need to wait 1 min to allow for unique dir names
+        print("Error: Please wait at least 1 minute between succesive runs")
+        print("Exiting...")
+        sys.exit(1)
+
+    for test in tests_dir.iterdir():
+        execute_dir, sub_file = create_test_execute_dir(timestamp_dir, test)
+
+        root_dir = os.getcwd()
+        schedd = htcondor2.Schedd()
+
+        job = generate_sub_object(sub_file, test.name)
+        item_data = [{"ResourceName": resource} for resource in resources.keys()]
+
+        job.issue_credentials()
+        os.chdir(execute_dir)
+        schedd.submit(job, itemdata=iter(item_data))
```

Comment:
```suggestion
        os.chdir(execute_dir)
        job = generate_sub_object(sub_file, test.name)
        item_data = [{"ResourceName": resource} for resource in resources.keys()]

        job.issue_credentials()
        schedd.submit(job, itemdata=iter(item_data))
```
We should `chdir()` before we start processing the submit file